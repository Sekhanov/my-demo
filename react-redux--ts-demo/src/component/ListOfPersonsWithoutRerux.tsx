import React from 'react'
import Person from '../data/Person'

interface IProps {
    persons : Person[];
    someString: string;
}


export default class ListOfPersonsWithoutRedux extends React.Component<IProps> {
    

    render() {
        const personList = this.props.persons.map(person => 
            (<li key={person.age}>name: {person.name} surname: {person.surname}</li>)        
        )

        return ( 
            <ul>
                { personList }
                <p className="ss"> {this.props.someString} </p>
            </ul>
            
        )
    }
}

