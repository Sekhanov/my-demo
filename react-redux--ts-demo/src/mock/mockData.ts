import Person from "../data/Person";

export const persons: Person[] = [
    new Person("ivan", "petrov", 10),
    new Person('peter', 'ivanov', 20),
    new Person('Sergey', 'KHanov', 3)
];

export const mockStoreData = {
    persons: [
      {
        name: 'ivan',
        surname: 'petrov',
        age: 10
      },
      {
        name: 'peter',
        surname: 'ivanov',
        age: 20
      },
      {
        name: 'Sergey',
        surname: 'KHanov',
        age: 3
      }
    ]
  }