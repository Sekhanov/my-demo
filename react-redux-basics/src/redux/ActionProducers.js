

export const processDataActionProducer = (data) => ({
    type: 'PROCESS_DATA',
    payload: data
})