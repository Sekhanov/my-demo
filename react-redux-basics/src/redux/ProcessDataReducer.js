import { persons } from "../data/Persons";

export default function processDataReducer(state = { data: persons }, action) {
    switch (action.type) {
        case 'PROCESS_DATA':
            return action.payload;
        default:
            return state;
    }   
}