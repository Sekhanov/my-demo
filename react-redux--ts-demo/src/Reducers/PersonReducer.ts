import Person from "../data/Person";
import { PersonsAction } from "../actions/ActionTypes";
import { PERSON_DATA } from "../constant/constants";
import { Persons } from "../mock/mockData";


const person = new Person('qwre', 'qwer', 1)


export const PersonReducer = (state: Person[] = Persons, action: PersonsAction): Person[] => {
    switch (action.type) {
        case PERSON_DATA:
           return  action.payload
        default:
            return state;
    }
}