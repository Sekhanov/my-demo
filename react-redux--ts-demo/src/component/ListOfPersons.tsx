import React from 'react'
import Person from '../data/Person'
import IGlobalState from '../store/IGlobalState'
import { connect } from 'react-redux'
import { Persons } from '../mock/mockData';


interface IProps {
    persons : Person[];
}


export class PersonList extends React.Component<IProps> {
    

    render() {
        const personList = this.props.persons.map(person => 
            (<li key={person.age}>name: {person.name} surname: {person.surname}</li>)        
        )

        return ( 
            <ul>
                { personList }
            </ul>

        )
    }
}


const mapStateToProps = (state: IGlobalState) => ({
    persons: state.persons
})


export default connect(
    mapStateToProps
)(PersonList)