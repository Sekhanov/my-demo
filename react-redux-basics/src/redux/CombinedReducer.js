import {combineReducers} from 'redux';
import processDataReducer from './ProcessDataReducer';
import searchStringReducer from './SearchStringReducer';



export  const combinedReducers = combineReducers(
    {
        data: processDataReducer,
        searchString: searchStringReducer
        }
    
)