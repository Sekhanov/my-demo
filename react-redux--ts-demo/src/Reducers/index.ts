import { PersonReducer } from "./PersonReducer";
import { combineReducers } from "redux";



export const allReducers = combineReducers({
    persons: PersonReducer
  });
  