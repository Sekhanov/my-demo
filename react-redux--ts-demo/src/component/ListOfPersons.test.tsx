import renderer from 'react-test-renderer'
import ListOfPersonsWithoutRerux from './ListOfPersonsWithoutRerux'
import React from 'react';
import { createStore } from 'redux';
import { PersonReducer } from '../Reducers/PersonReducer';
import { Persons } from '../mock/mockData';
import { ListOfPersons } from './ListOfPersons';


it("list of persons snapshot without redux", () => {
    const tree = renderer.create(
        <ListOfPersonsWithoutRerux
            persons={Persons}
            someString={'some string from props'}
        ></ListOfPersonsWithoutRerux>
    )
    expect(tree).toMatchSnapshot();
})

it('list of person snapshot apart from redux, import { ListOfPersons }', () => {
    const tree = renderer.create(
        <ListOfPersons
            persons={Persons}
            someString={'some string from props'}
        ></ListOfPersons>        
    )
    expect(tree).toMatchSnapshot();
})