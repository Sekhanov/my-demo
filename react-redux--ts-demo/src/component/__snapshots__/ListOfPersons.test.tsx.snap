// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list of person snapshot apart from redux, import { ListOfPersons } 1`] = `
<ul>
  <li>
    name: 
    ivan
     surname: 
    petronv
  </li>
  <li>
    name: 
    peter
     surname: 
    ivanov
  </li>
  <li>
    name: 
    Sergey
     surname: 
    KHanov
  </li>
  <p>
     
    some string from props
     
  </p>
</ul>
`;

exports[`list of persons snapshot without redux 1`] = `
<ul>
  <li>
    name: 
    ivan
     surname: 
    petronv
  </li>
  <li>
    name: 
    peter
     surname: 
    ivanov
  </li>
  <li>
    name: 
    Sergey
     surname: 
    KHanov
  </li>
  <p>
     
    some string from props
     
  </p>
</ul>
`;
